src/components/search-box/voice-search/VoiceSearchDialog.vue

```diff
@@ -0,0 +1,263 @@
+      }
+    }
+
+    socket.value.onerror = (error) => {
+      console.error('WebSocket error:', error)
+    }
+
+    socket.value.onclose = () => {
+      handleStopRecording()
+    }
+
+    const constraints = {
+      video: false,
+      audio: {
+        channelCount: 1,
+        echoCancellation: false,
+        sampleRate: 16000,
+      },
+    }
+
+    const stream = await navigator.mediaDevices.getUserMedia(constraints)
+    mediaStream.value = stream
+    mediaRecorder.value = new MediaRecorder(stream)
+    
+    mediaRecorder.value.onstart = startProgressBar
+    mediaRecorder.value.ondataavailable = onDataAvailableHandler
+    mediaRecorder.value.onstop = handleOnStopEvent
+    
+    // Send time slice every second
+    mediaRecorder.value.start(timeSliceLength)
+    isRecordingRef.value = true
+
+    setTimeout(() => {
+      if (isRecordingRef.value) {
+        stopRecognize()
+      }
+    }, timesliceLimit.value * timeSliceLength)
+  } catch (error) {
+    console.error('Error during recording start:', error)
+    return
+  }
+}
+
+const stopRecognize = () => {
+  handleStopRecording()
+
+  try {
+    socket.value?.send(JSON.stringify({ event: 'audio-chunk-end' }))
+    isRecordingRef.value = false
+
+    emitStoppedRecording()
+  } catch (error) {
+    console.error('Error during recording stop:', error)
+    return
+  }
+}
+
+const onDataAvailableHandler = (event: BlobEvent) => {
+  if (mediaRecorder.value?.state !== 'recording') {
+    return
+  }
+
+  // Encode audio data to base64 for sending over WebSocket
+  const reader = new FileReader()
+  reader.readAsDataURL(event.data)
+  reader.onloadend = () => {
+    const base64DataChunks = reader.result as string
+    const base64Data = base64DataChunks.split(',')[1]
+
+    socket.value?.send(JSON.stringify({ 
+      event: 'audio-chunk', 
+      data: base64Data 
+    }))
+  }
+}
+
+const handleStopRecording = () => {
+  if (!mediaRecorder.value || !mediaStream.value) {
+    return
+  }
+
+  if (
+    mediaRecorder.value?.state === 'inactive' && 
+    !isRecordingRef.value
+  ) {
+    return
+  }
+
+  try {
+    mediaRecorder.value?.stop()
+    mediaStream.value?.getTracks().forEach((track: MediaStreamTrack) => {
+      track.stop()
+    })
+  } catch (error) {
+    console.error('Error during recording stop:', error)
+    return
+  }
+}
+
+const emitStoppedRecording = (): void => {
+  emit('stop-recognize', transcription.value)
+}
+
+const handleDialogCloseEvent = (): void => {
+  emit('close')
+}
+
+const handleRecordingButtonClick = () => {
+  if (isRecordingRef.value) {
+    //dalay stop recording so that the last chunk is sent
+    setTimeout(() => {
+      stopRecognize()
+    }, stopDelay)
+  } else {
+    startRecognize()
+  }
+}
+
+const handleOnStopEvent = () => {
+  stopRecognize()
+}
+
+// Get the color of the progress bar, since it is set from the 
+// Lupa SCSS config var and not directly in the component
+const getProgressBarColor = (progressBarStyle: any) => {
```

Comment:
Maybe we could move progress bar and it's template to separate child component so this component would only handle voice search actions, and the progress bar component would handle display of the progress?